earthpy/plot.py

```diff
@@ -449,13 +460,25 @@ def hist(
         (<Figure size 800x300 with 3 Axes>, ...)
     """
 
+    if title:
+        if isinstance(title, str):
+            title = [title]
+
     # If the array is 3 dimensional setup grid plotting
     if arr.ndim > 2:
+        # Compress the arr if it's masked
+        if np.ma.is_masked(arr):
+            arrlis = []
+            for i in range(arr.shape[0]):
+                # Use compressed to flatten masked arr
+                arrlis.append(arr[i].compressed())
```

Comment:
here i used `compressed()` which is a method for masked_arrays. unfortunately you cant specify an axis or else this could be a `try` statement. i'm wondering if there is a way to run compressed on an array rather than looping. i'll ask on slack.